src/Action/CaptureAction.php

```diff
@@ -120,7 +120,7 @@ public function execute($request)
         $currencyCode = $payment->getCurrencyCode();
         $targetUrl = $request->getToken()->getTargetUrl();
         $amount = $payment->getAmount();
-        $transactionReference = "MercanetWS" . uniqid($payment->getOrder()->getNumber());
+        $transactionReference = "MercanetWS" . rand(100000,999999)."ORDER".$payment->getOrder()->getNumber();
```

Comment:
The rand function is not actually random. You'd better use uniqid(mt_rand(), true)). Also, please add spaces after the dots which follows our standards. This line should look like this:

"MercanetWS" . uniqid(mt_rand(), true)) . "ORDER" . $payment->getOrder()->getNumber();